pymeasure/instruments/teledyne/teledyneT3AFG.py

```diff
@@ -0,0 +1,153 @@
+        validator=strict_discrete_set,
+        values=['SINE', 'SQUARE', 'RAMP', 'PULSE', 'NOISE', 'ARB', 'DC', 'PRBS', 'IQ'],
+        get_process=lambda x: x[1],
+    )
+
+    # TODO: Add frequency ranges per model?
+    frequency = Channel.control(
+        "C{ch}:BSWV?",
+        "C{ch}:BSWV FRQ,%g",
+        """Control the frequency of waveform to be output in Hertz.
+        Has no effect when WVTP is NOISE or DC.""",
+        validator=strict_range,
+        values=[0, 350e6],
+        get_process=get_process_generator('FRQ', 'HZ', float),
+        dynamic=True,
+        check_set_errors=True,
+    )
+
+    # TODO: Add range of outputs per model? Tricky without knowing offset and frequency
+    amplitude = Channel.control(
+        "C{ch}:BSWV?",
+        "C{ch}:BSWV AMP,%g",
+        """Control the amplitude of waveform to be output in volts peak-to-peak.
+        Has no effect when WVTP is NOISE or DC.
+        Max amplitude depends on offset, frequency, and load.
+        Amplitude is also limited by the channel max output amplitude.""",
+        validator=strict_range,
+        values=[0, 5],
+        get_process=get_process_generator('AMP', 'V', float),
+        dynamic=True,
+        check_set_errors=True,
+    )
+
+    # TODO: Add range of offset per model? Tricky without knowing amplitude, frequency
+    offset = Channel.control(
+        "C{ch}:BSWV?",
+        "C{ch}:BSWV OFST,%g",
+        """Control the offset of waveform to be output in volts.
+        Has no effect when WVTP is NOISE.
+        Max offset depends on amplitude, frequency, and load.
+        Offset is also limited by the channel max output amplitude.""",
+        validator=strict_range,
+        values=[0, 5],
+        get_process=get_process_generator('OFST', 'V', float),
+        dynamic=True,
+        check_set_errors=True,
+    )
+
+    # TODO: Add other Basic Waveform parameters like period
+
+    max_output_amplitude = Channel.control(
+        "C{ch}:BSWV?",
+        "C{ch}:BSWV MAX_OUTPUT_AMP,%g",
+        """Control the maximum output amplitude of the channel in volts peak to peak.""",
+        validator=strict_range,
+        values=[0,20],
+        get_process=get_process_generator('MAX_OUTPUT_AMP', 'V', float),
+        dynamic=True,
+    )
+
+
+class TeledyneT3AFG(Instrument):
+    """Represents the Teledyne T3AFG series of arbitrary waveform
+    generator interface for interacting with the instrument.
+
+    Intially targeting T3AFG80, some features may not be avalible on
+    lower end models and features from higher end models are not
+    included here intially.
+
+    .. code-block: python
+    generator=TeledyneT3AFG(resource)
+    generator.ch_1.wavetype='SINE'
+    generator.ch_1.amplitude=2
+    generator.ch_1.output_enabled=True
+    """
+
+    channels = Instrument.ChannelCreator(SignalChannel, (1, 2))
+
+    def __init__(self, adapter, name="Teledyne T3AFG", **kwargs):
+        super().__init__(
+            adapter, name, **kwargs
```

Comment:
Does this instrument use default SCPI commands? If so, could you please add it explicitly:
```suggestion
            adapter, name, includeSCPI=True, **kwargs
```

Reply:
The programming manual doesn't include anything obviously claiming it has all of the standard SCPI commands and doesn't list most of them, but they all seem to work so I'll add this.